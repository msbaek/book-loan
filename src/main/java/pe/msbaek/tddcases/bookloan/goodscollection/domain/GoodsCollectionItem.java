package pe.msbaek.tddcases.bookloan.goodscollection.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Comment;

@NoArgsConstructor
@Getter
@Entity
@Table(name = "GOODS_COLLECTION_ITEM")
@Comment("특정 상품군 아이템")
public class GoodsCollectionItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Comment("특정 상품군 아이템 ID")
    private Long id;
    @Column(name = "GOODS_NO")
    @Comment("상품 No")
    private Long goodsNo;
    @Column(name = "GOODS_ID")
    @Comment("상품 ID")
    private String goodsId;
    @Column(name = "BARCODE")
    @Comment("상품 바코드")
    private String barcode;
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "GOODS_COLLECTION_ID", nullable = false)
    private GoodsCollection goodsCollection;
    public static GoodsCollectionItem of(Goods goods) {
        GoodsCollectionItem item = new GoodsCollectionItem();
        item.goodsNo = goods.goodsNo();
        item.goodsId = goods.goodsId();
        item.barcode = goods.barcode();
        return item;
    }

    public void setId(Long id) {
        this.id = id;
    }
}