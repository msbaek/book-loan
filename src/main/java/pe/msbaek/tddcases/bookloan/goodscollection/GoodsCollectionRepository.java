package pe.msbaek.tddcases.bookloan.goodscollection;

import org.springframework.data.domain.Pageable;

import java.util.List;

public interface GoodsCollectionRepository {
    List<GoodsCollection> findByNameContaining(String name, Pageable pageable);
}
