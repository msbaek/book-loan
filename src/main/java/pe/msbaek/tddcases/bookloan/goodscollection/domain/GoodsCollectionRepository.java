package pe.msbaek.tddcases.bookloan.goodscollection.domain;

import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface GoodsCollectionRepository {
    List<GoodsCollection> findByNameContaining(String name, Pageable pageable);
    Optional<GoodsCollection> findById(Long id);

    List<Goods> listGoodsByIds(List<String> ids);

    GoodsCollection save(GoodsCollection goodsCollection);
}