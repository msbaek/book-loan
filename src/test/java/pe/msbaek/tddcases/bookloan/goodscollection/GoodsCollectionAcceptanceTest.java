package pe.msbaek.tddcases.bookloan.goodscollection;

import com.ktown4u.utils.Neutralizer;
import com.ktown4u.utils.YamlPrinter;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.graphql.tester.AutoConfigureHttpGraphQlTester;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.graphql.test.tester.GraphQlTester;
import org.springframework.graphql.test.tester.HttpGraphQlTester;
import org.springframework.test.context.jdbc.Sql;
import pe.msbaek.tddcases.bookloan.goodscollection.domain.GoodsCollectionRepository;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

// @ActiveProfiles("test")
@AutoConfigureHttpGraphQlTester
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
public class GoodsCollectionAcceptanceTest {
    @Autowired private HttpGraphQlTester graphQlTester;
    @Autowired private GoodsCollectionRepository goodsCollectionRepository;

    @Test
    @Sql("classpath:GOODS.sql")
    public void query_pagedGoodsCollection() throws Exception {
        String queryString = """
                query {
                  pagedGoodsCollection(request: {
                    keyword: "name",
                    type: "type",
                    page: 0,
                    size: 10
                  }) {
                    content {
                      id
                      name
                      createdBy
                      createdAt
                      goodsCollectionItems {
                        goodsNo
                        goodsId
                        barcode
                      }
                    }
                  }
               }
                """;

        List<GetGoodsCollection.GoodsCollectionDto> result = request(queryString, "pagedGoodsCollection.content")
                .entityList(GetGoodsCollection.GoodsCollectionDto.class)
                .get();
        Approvals.verify(Neutralizer.localDateTime(YamlPrinter.printWithExclusions(result, "updatedBy" ,"updatedAt")));
    }

    @Test
    @Sql("classpath:GetGoodsCollectionsTest.sql")
    public void create_goods_collection() throws Exception {
        String queryString = """
                mutation {
                    createGoodsCollection(request: {
                        name: "상품군 이름"
                        ids: ["GD00111839", "GD00111838", "GD00111836"]
                    })
                }
                """;

        Long result = request(queryString, "createGoodsCollection")
                .entity(Long.class)
                .get();
        assertThat(result).isGreaterThan(0L);
    }

    private GraphQlTester.Path request(String queryString, String requestName) {
        return this.graphQlTester
                .mutate()
                .header("x-tester-no", "1001")
                .build()
                .document(queryString)
                .execute()
                .path(requestName);
    }
}