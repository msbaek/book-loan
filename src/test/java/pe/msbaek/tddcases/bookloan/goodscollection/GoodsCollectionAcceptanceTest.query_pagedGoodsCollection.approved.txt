---
- id: 1
  name: "name1"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111838
    goodsId: "GD00111838"
    barcode: "9000000111838"
  - goodsNo: 111838
    goodsId: "GD00111838"
    barcode: "9000000111838"
  - goodsNo: 111838
    goodsId: "GD00111838"
    barcode: "9000000111838"
- id: 2
  name: "name2"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111836
    goodsId: "GD00111836"
    barcode: "8809973502397"
  - goodsNo: 111836
    goodsId: "GD00111836"
    barcode: "8809973502397"
  - goodsNo: 111836
    goodsId: "GD00111836"
    barcode: "8809973502397"
- id: 3
  name: "name3"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111835
    goodsId: "GD00111835"
    barcode: "8809973502397"
  - goodsNo: 111835
    goodsId: "GD00111835"
    barcode: "8809973502397"
  - goodsNo: 111835
    goodsId: "GD00111835"
    barcode: "8809973502397"
- id: 4
  name: "name4"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111834
    goodsId: "GD00111834"
    barcode: "8809973502397"
  - goodsNo: 111834
    goodsId: "GD00111834"
    barcode: "8809973502397"
  - goodsNo: 111834
    goodsId: "GD00111834"
    barcode: "8809973502397"
- id: 5
  name: "name5"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111833
    goodsId: "GD00111833"
    barcode: "8809973502397"
  - goodsNo: 111833
    goodsId: "GD00111833"
    barcode: "8809973502397"
  - goodsNo: 111833
    goodsId: "GD00111833"
    barcode: "8809973502397"
- id: 6
  name: "name6"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111832
    goodsId: "GD00111832"
    barcode: "8809973502397"
  - goodsNo: 111832
    goodsId: "GD00111832"
    barcode: "8809973502397"
  - goodsNo: 111832
    goodsId: "GD00111832"
    barcode: "8809973502397"
- id: 7
  name: "name7"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111831
    goodsId: "GD00111831"
    barcode: "8809973502397"
  - goodsNo: 111831
    goodsId: "GD00111831"
    barcode: "8809973502397"
  - goodsNo: 111831
    goodsId: "GD00111831"
    barcode: "8809973502397"
- id: 8
  name: "name8"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111830
    goodsId: "GD00111830"
    barcode: "8809973502397"
  - goodsNo: 111830
    goodsId: "GD00111830"
    barcode: "8809973502397"
  - goodsNo: 111830
    goodsId: "GD00111830"
    barcode: "8809973502397"
- id: 9
  name: "name9"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111826
    goodsId: "GD00111826"
    barcode: "8809966901251"
  - goodsNo: 111826
    goodsId: "GD00111826"
    barcode: "8809966901251"
  - goodsNo: 111826
    goodsId: "GD00111826"
    barcode: "8809966901251"
- id: 10
  name: "name10"
  createdBy: 1
  createdAt: "2003-05-03T10:11:12.134567"
  goodsCollectionItems:
  - goodsNo: 111839
    goodsId: "GD00111839"
    barcode: "8809888410251"
  - goodsNo: 111839
    goodsId: "GD00111839"
    barcode: "8809888410251"
  - goodsNo: 111839
    goodsId: "GD00111839"
    barcode: "8809888410251"
